### ISSUE TITLE
`control_statement` asks to illegally unwrap guard.

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#5846]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`control_statement` checks to ensure the control statement is not unnecessarily wrapped in parenthesis. However, parenthesis are required in certain cases for control statements, such as when the statement itself consists of braces, to disambiguate the braces from the control operator's braces. The following example illustrates where `control_statement` demands the removal of parentheses which would render the statement illegal in the language.

```swift
guard ({ switch target {
    case a: true
    case b: false
}
}())
else { return false }
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/Users/hubstaff/Developer/hubstaff/ios/test.swift:1:1: warning: Control Statement Violation: `if`, `for`, `guard`, `switch`, `while`, and `catch` statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses (control_statement)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure): 0.57.0
* Xcode version (run `xcodebuild -version` to be sure): 16.1, 16B40
* Installation method used (Homebrew, CocoaPods, building from source, etc): brew
